runtime/src/accounts.rs

```diff
@@ -635,26 +635,15 @@ impl Accounts {
         }
     }
 
-    fn filter_zero_lamport_account(
-        account: AccountSharedData,
-        slot: Slot,
-    ) -> Option<(AccountSharedData, Slot)> {
-        if account.lamports() > 0 {
-            Some((account, slot))
-        } else {
-            None
-        }
-    }
-
     /// Slow because lock is held for 1 operation instead of many
+    /// This always returns None for zero-lamport accounts.
     fn load_slow(
         &self,
         ancestors: &Ancestors,
         pubkey: &Pubkey,
         load_hint: LoadHint,
     ) -> Option<(AccountSharedData, Slot)> {
-        let (account, slot) = self.accounts_db.load(ancestors, pubkey, load_hint)?;
-        Self::filter_zero_lamport_account(account, slot)
+        self.accounts_db.load(ancestors, pubkey, load_hint)
```

Comment:
I hacked this up to cause this function to return Some(empty), in case we were to accidentally do this now.
These 5 tests failed indicating we are testing this behavior:
```
    bank::tests::test_bank_squash
    bank::tests::test_drained_created_account
    bank::tests::test_invalid_rent_state_changes_existing_accounts
    bank::tests::test_purge_empty_accounts
    snapshot_utils::tests::test_incremental_snapshots_handle_zero_lamport_accounts```